mysql-test/suite/rpl/r/rpl_create_select_row.result

```diff
@@ -0,0 +1,158 @@
+master-bin.000001	#	Query	#	#	ROLLBACK
+#
+# Engine = myisam
+#
+set statement binlog_format=statement for create table t (a int) select 1 as a;
+set statement binlog_format=row for create or replace table t (a int primary key, b char) engine=myisam select 1 AS a, 'b' as b union select 1 as a, 'c' as b;
+ERROR 23000: Duplicate entry '1' for key 'PRIMARY'
+select * from t;
+ERROR 42S02: Table 'test.t' doesn't exist
+#
+# Prove an expected lonely `DROP table t'
+include/show_binlog_events.inc
+Log_name	Pos	Event_type	Server_id	End_log_pos	Info
+master-bin.000001	#	Gtid	#	#	BEGIN GTID #-#-#
+master-bin.000001	#	Query	#	#	use `test`; DROP TABLE IF EXISTS `test`.`t`/* Generated to handle failed CREATE OR REPLACE */
+master-bin.000001	#	Query	#	#	ROLLBACK
+set statement binlog_format=statement for create table t (a int) select 1 as a;
+set statement binlog_format=row for create or replace table t (a text) engine=myisam select repeat ('a',1024) AS a union select repeat ('a',3*4096) AS a union select repeat ('a',3*4096) AS a;
+ERROR HY000: Multi-statement transaction required more than 'max_binlog_cache_size' bytes of storage; increase this mariadbd variable and try again
+select * from t;
+ERROR 42S02: Table 'test.t' doesn't exist
+#
+# Prove an expected lonely `DROP table t'
+include/show_binlog_events.inc
+Log_name	Pos	Event_type	Server_id	End_log_pos	Info
+master-bin.000001	#	Gtid	#	#	BEGIN GTID #-#-#
+master-bin.000001	#	Query	#	#	use `test`; DROP TABLE IF EXISTS `test`.`t`/* Generated to handle failed CREATE OR REPLACE */
+master-bin.000001	#	Query	#	#	ROLLBACK
+set statement binlog_format=statement for create table t (a int) select 1 as a;
+set statement binlog_format=row for create or replace table t (a text) engine=myisam select repeat ('a',4096*3) AS a;;
+ERROR HY000: Multi-statement transaction required more than 'max_binlog_cache_size' bytes of storage; increase this mariadbd variable and try again
+select * from t;
+ERROR 42S02: Table 'test.t' doesn't exist
+#
+# Prove an expected lonely `DROP table t'
+include/show_binlog_events.inc
+Log_name	Pos	Event_type	Server_id	End_log_pos	Info
+master-bin.000001	#	Gtid	#	#	BEGIN GTID #-#-#
+master-bin.000001	#	Query	#	#	use `test`; DROP TABLE IF EXISTS `test`.`t`/* Generated to handle failed CREATE OR REPLACE */
+master-bin.000001	#	Query	#	#	ROLLBACK
+create table ti_pk (a int primary key) engine=innodb;
+create table ta (a int) engine=aria;
+create function f_ia(arg int)
+returns integer
+begin
+insert into ti_pk set a=1;
+insert into ta set a=1;
+insert into ti_pk set a=arg;
+return 1;
+end |
+set statement binlog_format = ROW for create table t_y (a int) engine=aria select f_ia(1 /* err */) as a;
+ERROR 23000: Duplicate entry '1' for key 'PRIMARY'
+select * from t_y;
+ERROR 42S02: Table 'test.t_y' doesn't exist
+# correct execution: `ta` is modified and its new record is binlogged
+include/show_binlog_events.inc
+Log_name	Pos	Event_type	Server_id	End_log_pos	Info
```

Comment:
I'm a bit confused here, this looks like correct behavior to me, but the JIRA description in MDEV-36027 says this is a bug. That is, it looks like all data-modifying information is binlogged. The `create table t_y...` isn't binlogged, but that seems correct, as the query errors and the table creation is rolled back. The updates to `ti_pk` are rolled back, and thereby not binlogged. The only thing binlogged is the update to `ta`, which is the only thing that changed data (by my understanding).

Reply:
In case `--error ER_DUP_ENTRY` MDEV-36027 requires the statement format (as the description exemplifies). Here it is ROW.

Reply:
Hmm, but we set `set statement binlog_format = ROW`. And with this configuration, don't we re-write the DDL to exclude the `SELECT` component from the user input `CREATE TABLE ... SELECT` query at binlogging, and then provide the data changes as row events? And thus, when the query fails and everything that can be rolled back, is, and the items that are permanent are logged in `ROW` format.

Reply:
Just in case, I should've said ^ that YES at this point the execution is correct.

Now to your followup, I don't see any disagreement with what I tried to convey..

In the ROW the `ER_DUP_ENTRY` failed **at "selecting"** `C..S` always knows how to clean up.
So we have only those
>  the items that are permanent are logged in ROW format
```include/show_binlog_events.inc
Log_name	Pos	Event_type	Server_id	End_log_pos	Info
master-bin.000001	#	Gtid	#	#	BEGIN GTID #-#-#
master-bin.000001	#	Table_map	#	#	table_id: # (test.ta)
master-bin.000001	#	Write_rows_v1	#	#	table_id: # flags: STMT_END_F
```